mycore-sass/src/main/java/org/mycore/sass/MCRResourceImporter.java

```diff
@@ -0,0 +1,142 @@
+
+    private static final UrlHelper URL_HELPER = new UrlHelper();
+
+    /**
+     * Initializes a new instance of the MCRResourceImporter class.
+     *
+     * @param servletContext The ServletContext used for importing resources.
+     */
+    public MCRResourceImporter(ServletContext servletContext) {
+        this.servletContext = servletContext;
+    }
+
+    /**
+     * Canonicalizes the given URL of the import and determines if it is from an `@import` rule.
+     * <br>
+     * This method takes a URL as input and checks if it starts with the "sass:/" prefix. If it does,
+     * it removes the prefix and attempts to locate the resource URL using the getWebResourceUrl() method. If the
+     * resource URL is found and is a file, it returns the original URL; otherwise, it returns null
+     *
+     * @param url The URL of the import to be canonicalized. This may be either absolute or relative.
+     * @param fromImport Whether this request comes from an `@import` rule.
+     * @return The canonicalized URL if it starts with "sass:/" and the resource exists, otherwise null.
+     * @throws Exception If an error occurs during the canonicalization process.
+     */
+    @Override
+    public String canonicalize(String url, boolean fromImport) throws Exception {
+        LOGGER.debug(() -> "handle canonicalize: " + url);
+        if (!url.startsWith(SASS_URL_PREFIX)) {
+            return null;
+        }
+        String resourcePath = url.substring(SASS_URL_PREFIX.length());
+        URL resource = getWebResourceUrl(resourcePath);
+        if (resource == null || resource.getFile().endsWith("/") || !URL_HELPER.isFile(resource)) {
+            //with jetty the directory does not end with /, so we need to REAL check
+            return null;
+        }
+        LOGGER.debug("Resolved {} to {}", url, resource);
+        return resource == null ? null : url;
+    }
+
+    /**
+     * Handles the import of a URL and returns the success of the import along with its content.
+     *
+     * This method takes a URL as input and checks if it starts with the "sass:/" prefix. If it does,
+     * it attempts to locate the resource URL. If the resource URL is found,
+     * it reads its content returns the an ImportSuccess object.
+     *
+     * @param url The URL of the resource to import.
+     * @return An ImportSuccess object containing the content and syntax of the imported resource.
+     * @throws IOException If there is an error in handling the import.
+     */
+    @Override
+    public ImportSuccess handleImport(String url) throws IOException {
+        LOGGER.debug(() -> "handle import: " + url);
+        if (!url.startsWith(SASS_URL_PREFIX)) {
+            return null;
+        }
+        ImportSuccess.Builder result = ImportSuccess.newBuilder();
+        String resourcePath = url.substring(SASS_URL_PREFIX.length());
+        LOGGER.debug(() -> "resource: " + resourcePath);
+        URL configResource = getWebResourceUrl(resourcePath);
```

Comment:
Seems to me like 

```java
        if (!url.startsWith(SASS_URL_PREFIX)) {
            return null;
        }
        String resourcePath = url.substring(SASS_URL_PREFIX.length());
        LOGGER.debug(() -> "resource: " + resourcePath);
        URL configResource = getWebResourceUrl(resourcePath);
        LOGGER.debug(() -> "resource url: " + configResource);
```
(and possible the special handling of Jettys behaviors) should be a common method used by `canonicalize` as well as `handleImport`. 

The necessity for the following differences isn't immediately obvious and I'd guess that there actually is no such necessity.

- Whats the significance of naming the result of `getWebResourceUrl` differently in the two methods. Is there actually a difference?
- Why is the logging only relevant in `handleImport`?
- Why is the special handling of Jettys behaviors not present in `handleImport`? Is there a difference or is this an oversight?